apps/src/maze/beeCell.js

```diff
@@ -0,0 +1,178 @@
+/**
+ * @overview Cell represents the contets of the grid elements for Bee.
+ * Bee Cells are more complex than many other kinds of cell; they can be
+ * "hidden" with clouds, they can represent multiple different kinds of
+ * element (flower, hive), some of which can be multiple colors (red,
+ * purple), and which can have a range of possible values.
+ *
+ * Some cells can also be "variable", meaning that their contents are
+ * not static but can in fact be randomized between runs.
+ */
+
+// FC is short for FlowerComb, which we were originally using instead of cloud
+var CLOUD_MARKER = 'FC';
+
+var Cell = function (value, clouded, prefix, color) {
+  /**
+   * @type {String}
+   */
+  this.clouded_ = clouded;
+
+  /**
+   * @type {String}
+   */
+  this.prefix_ = prefix;
+
+  /**
+   * @type {String}
+   */
+  this.color_ = color;
+
+  /**
+   * @type {Number}
+   */
+  this.originalValue_ = value;
+
+  /**
+   * @type {Number}
+   */
+  this.currentValue_ = undefined;
+  this.resetCurrentValue();
+};
+
+module.exports = Cell;
+
+/**
+ * Returns a new Cell that's an exact replica of this one
+ * @return {Cell}
+ */
+Cell.prototype.clone = function () {
+  var newCell = new Cell(this.originalValue_, this.clouded_, this.prefix_, this.color_);
+  newCell.setCurrentValue(this.currentValue_);
+  return newCell;
+};
+
+/**
+ * @return {Number}
+ */
+Cell.prototype.getCurrentValue = function () {
+  return this.currentValue_;
+};
+
+/**
+ * @param {Number}
+ */
+Cell.prototype.setCurrentValue = function (val) {
+  this.currentValue_ = val;
+};
+
+Cell.prototype.resetCurrentValue = function () {
+  if (this.prefix_) {
+    this.currentValue_ = parseInt(this.prefix_ + this.originalValue_);
+  } else {
+    this.currentValue_ = 0;
+  }
+};
+
+Cell.prototype.isFlower = function () {
+  return this.prefix_ === '+';
+};
+
+Cell.prototype.isHive = function () {
+  return this.prefix_ === '-';
+};
+
+Cell.prototype.isClouded = function () {
```

Comment:
Seems a little surprising that isClouded returns true for only one particular cloud type (FC). Should this be named something else? Or maybe at least a comment explaining